CMA-19 Add messages functionality

weronikac opened the pull request:
# add endpoints for messages functionality

AleksandraRolka committed:
CMA-58 general frontend layout and structure - part 1

AleksandraRolka committed:
CMA-21 extend tags exceptions handling

AleksandraRolka committed:
CMA-21 extend tags exceptions handling

AleksandraRolka committed:
CMA-58  general frontend layout and structure - part 2, CMA-37 create tags functionality

weronikac committed:
CMA-19 Add messages functionality
# add endpoints for messages functionality

weronikac committed:
CMA-19 Add messages functionality
# add endpoints for messages functionality

weronikac committed:
CMA-19 Add messages functionality

weronikac committed:
CMA-19 Add messages functionality
# rebase

weronikac committed:
CMA-19 Add messages functionality

BlackDilvish commented on server/src/main/java/com/project/cisco/controler/MessageApiController.java:
```diff
@@ -0,0 +1,49 @@
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
```
Let's add here origin to fix an error releated to CORS on frontend

`@CrossOrigin(value = { "http://localhost:3000/" },
        maxAge = 900
)`

weronikac replied:
done

BlackDilvish commented on server/src/main/java/com/project/cisco/service/impl/MessageServiceImpl.java:
```diff
@@ -0,0 +1,103 @@
+    private MessageMapper messageMapper;
+
+    @Override
+    public MessageDto addMessage(MessageDto messageDto) {
```
We have a use case specifying that: "Translation should have same tags as original message". I believe that should be on a backend side, to check if `original_message` has any tags and copy them to the new one.

AleksandraRolka commented:
@weronikac please add basic requests for messages functionalities in:
src/test/java/com/project/cisco/postman/**cisco-app.postman_collection.json**

AleksandraRolka commented:
You should also take care of this case

![image](https://user-images.githubusercontent.com/56655867/211196579-7454da82-b8ba-4ac9-9c54-1c9ac6373f1a.png)
because now it does not work like that, it throws exception:
![image](https://user-images.githubusercontent.com/56655867/211196744-933404d3-4132-4db2-bb2d-c6946ec3a73b.png)


please check also others requirements for messages if are fulfilled

AleksandraRolka commented:
If on backend Deleting, Modifying English will be disabled you could remove this options in Operations column for that language

weronikac committed:
CMA-19 Add messages functionality
# message has to have same tags as original message
# fix modifying messages

weronikac committed:
CMA-19 Add messages functionality
# fix modifying original messages

weronikac committed:
CMA-19 Add messages functionality
# if a language is deleted, all the messages that had it as “language” should be deleted too
# f a tag is deleted, all the messages that had it as “tag” should modify field “tags” to not have deleted tag in the field

weronikac committed:
CMA-19 Add messages functionality
# add postman tests

weronikac committed:
CMA-19 Add messages functionality
# copy tags from original message

weronikac committed:
Merge branch 'master' into CMA-19-Add-messages-functionality

weronikac committed:
CMA-19 Add messages functionality
# rebase
# add English language when starting app
# disable deleting English language
